tests/e2e/pageobjects/dashboard/CreateWorkspace.ts

```diff
@@ -14,53 +14,95 @@ import { DriverHelper } from '../../utils/DriverHelper';
 
         const sampleLocator: By = this.getSampleLocator(sampleName);
 
         await this.driverHelper.waitAndClick(sampleLocator, timeout);
     }
 
-    async startWorkspaceUsingFactory(factoryUrl: string, timeout: number = TimeoutConstants.TS_CLICK_DASHBOARD_ITEM_TIMEOUT) {
+    async startWorkspaceUsingFactory(factoryUrl: string, timeout: number = TimeoutConstants.TS_CLICK_DASHBOARD_ITEM_TIMEOUT): Promise<void> {
         Logger.debug(`CreateWorkspace.startWorkspaceUsingFactory factoryUrl: "${factoryUrl}"`);
         await this.driverHelper.waitVisibility(CreateWorkspace.FACTORY_URL_LOCATOR, timeout);
         await this.driverHelper.type(CreateWorkspace.FACTORY_URL_LOCATOR, Key.chord(factoryUrl, Key.ENTER), timeout);
     }
 
+    async clickOnSampleForSpecificEditor(sampleName: string, timeout: number = TimeoutConstants.TS_CLICK_DASHBOARD_ITEM_TIMEOUT): Promise<void> {
+        await this.clickOnEditorsDropdownListButton(sampleName, timeout);
+
+        Logger.debug(`CreateWorkspace.clickOnSampleForSpecificEditor sampleName: "${sampleName}"`);
+
+        const sampleLocator: By = this.getSampleLocatorWithSpecificEditor(sampleName);
+        await this.driverHelper.waitAndClick(sampleLocator, timeout);
+    }
+
+    private async clickOnEditorsDropdownListButton(sampleName: string, timeout: number): Promise<void> {
+        Logger.debug(`CreateWorkspace.clickOnSample sampleName: "${sampleName}, editor ${TestConstants.TS_SELENIUM_EDITOR}"`);
+
+        const editorDropdownListLocator: By = this.getEditorsDropdownListLocator(sampleName);
+        await this.driverHelper.waitAndClick(editorDropdownListLocator, timeout);
+    }
+
+    private getEditorsDropdownListLocator(sampleName: string): By {
+        return By.xpath(`//div[text()=\'${sampleName}\']//parent::article//button`);
+    }
+
+    private getSampleLocatorWithSpecificEditor(sampleName: string): By {
+        let editor: string = '';
+        switch (TestConstants.TS_SELENIUM_EDITOR) {
+            case EditorType.CHE_CODE:
+                editor = 'code';
+                break;
+            case EditorType.THEIA:
+                editor = 'theia';
+                break;
+            default:
+                throw new Error(`Unsupported editor ${TestConstants.TS_SELENIUM_EDITOR}`);
+        }
+
+        Logger.debug(`CreateWorkspace.getSampleLocator sampleName: ${sampleName}, editor "${editor}"`);
+
+        return By.xpath(`//div[text()='${sampleName}']//parent::article//span[text()[
+                contains(
+                translate(., 'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz'),
+                '${editor}')]
+            ]//parent::a`);
+    }
+
     private getSampleLocator(sampleName: string): By {
-        Logger.trace(`CreateWorkspace.getSampleLocator sampleName: ${sampleName}`);
+        Logger.debug(`CreateWorkspace.getSampleLocator sampleName: ${sampleName}, used default editor`);
```

Comment:
Please move the "using default editor" log message into `clickOnSample` method instead and switch it back to trace to not clutter the logs

Reply:
fixed